### PR TITLE
Fix ens resolution in "start new chat" according to latest dapp developments

### PR DESCRIPTION
**Pull request by yenda**
### Summary:

Add support for pubkey method on ens contract

https://ipfs.infura.io/ipfs/QmdpwmuM3UN7Zb93QYKeMobHqpyWevNzL9nYxK7S3cYJbS/
In latest version of the name registration dapp for stateofus.eth names, the pubkey is now stored using X and Y coordinates on the elliptic curve. To resolve it we must now use the pubkey method instead of the text method

### Steps to test:
- create a name in the dapp on ropsten (need eth and stt) https://ipfs.infura.io/ipfs/QmdpwmuM3UN7Zb93QYKeMobHqpyWevNzL9nYxK7S3cYJbS/
- change account, try to add contact using registered user name

status: ready 

<blockquote><div><strong><a href="https://ipfs.infura.io/ipfs/QmdpwmuM3UN7Zb93QYKeMobHqpyWevNzL9nYxK7S3cYJbS/">Status.im Domain Registration</a></strong></div></blockquote>

### CODE REVIEW

**Review by goranjovic — APPROVED** [review]

**Review by flexsurfer — APPROVED** [review]

File: src/status_im/utils/ethereum/ens.cljs
Comment (jeluard) [inline src/status_im/utils/ethereum/ens.cljs]: Can we add comment to this magic / extract it as a named fn?

**Review by jeluard — APPROVED** [review]

**Review by bgits — APPROVED** [review]
LGTM

**Comment by lukaszfryc** [comment]
@yenda do you know how long it may take to register a name? I've already waited for ~10 min and I still see the spinner:



[Here](https://ropsten.etherscan.io/tx/0x4c5cf2ddc8e3b076e8ef6f81a77525cced9e457a5503965484c6fe1a4e8a91f1) is the last transaction I did (`Function: approve(address _spender, uint256 _amount)`). It's already confirmed.

**Comment by lukaszfryc** [comment]
@yenda when I tested 2nd time on iOS the spinner disappeared after few seconds (the `approve` function was still pending in etherscan). But, when I then tried to click the register button it appeared again and again so I could not complete the ens registration.

**Comment by yenda** [comment]
@lukaszfryc the dapp is not related to this because it's not in the repo, I also had a bit of trouble to register the username have you tried refreshing the page then do the process again ?
I have the username `qweqwe.stateofus.eth` registered already in case you don't succeed in registering.
The PR is really just about resolving those names correctly not registering them

**Comment by lukaszfryc** [comment]
Let me try again as well as trying your username

**Comment by lukaszfryc** [comment]
@yenda I still have issues registering my username but yours works fine!

**Pull request merged**
